build.gradle

```diff
@@ -380,6 +430,7 @@ plugins.withType(JavaPlugin) {
   tasks.javadoc.options.links "http://typesafehub.github.io/config/latest/api/",
                               "https://docs.oracle.com/javase/7/docs/api/",
                               "http://docs.guava-libraries.googlecode.com/git-history/v15.0/javadoc/"
+  rootProject.ext.javadocLinks.each{tasks.javadoc.options.links it}
```

Comment:
For external libraries, there is no way we can add those programatically. We have to add those by hand.

For internal, it should be possible to figure out the links from projects in dependencies but it is a bit trickier. I'll leave that for V.next;